app/src/main/java/org/dharmaseed/android/TalkRepository.java

```diff
@@ -31,37 +31,42 @@ public Cursor getTalkAdapterData(
             boolean isDownloaded
     )
     {
-        List<String> columns = new ArrayList<>();
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.ID);
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TITLE);
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TEACHER_ID);
-        columns.add(DBManager.C.Teacher.TABLE_NAME + "." + DBManager.C.Teacher.NAME);
-        return getTalks(columns, searchTerms, isStarred, isDownloaded);
+
+        return getTalks(searchTerms, isStarred, isDownloaded, "");
     }
 
     /**
-     * @param columns the columns to select or null if all
      * @param searchTerms the term to filter by or null if none
      * @param isStarred whether the talk is starred
      * @param isDownloaded whether the talk is downloaded
+     * @param extraWhere an extra where clause that can be used to filter on a specific teacher/center id
      * @return every talk in the db filtered by search term, is starred, and is downloaded
      */
     private Cursor getTalks(
-           List<String> columns,
```

Comment:
I think these should be left here because `getTalks()` is about retrieving talk data where the caller can pass in the data needed. It's a general function where calling methods can be more specific. For example, if - in the future - we want a query that only gives us the `audio_url` of every talk, we just create a new method that calls `getTalks()` and passes in that single column. The way you've made it, we would have to add `audio_url` to _every_ talk query in the whole program (by adding it to `getTalks()`), which would then go unused everywhere except where we needed it, wasting time and space.

Reply:
Thanks, I have a better understanding now of what you were intending here, and that makes a lot of sense.  I'll add `columns` back in! :)